### ISSUE TITLE
Proposal: numbering all section levels

### ISSUE THREAD

**Issue by rswgnu** [p3r7/awesome-elisp#19]
This is fantastic.  Thanks for the effort.

I would suggest numbering the sections so people can refer to them.

**Comment by p3r7** [comment]
Hi

Indeed, that would help, but there are 2 arguments against it:
 - the document is still rapidly evolving and we cannot guarantee that a numeric reference will remain valid for a long time
 - we follow the style of other [awesome lists](https://github.com/sindresorhus/awesome), which doesn't seem to favor numbering

An alternative when it comes to referring a section is simply link to it. Each section header provides an anchor.
E.g.`Development Tools > Code Editing > Jump to definition > elisp-def`can be shared (and referred to) as https://github.com/p3r7/awesome-elisp#elisp-def.

There is just one limitation: html anchors have no notion of hierarchy (IMHO, they did a bad design on this).
We currently have an issue w/ `elisp-demos` which is listed under both `Resources > Cookbooks` and `Development Tools > Documentation & Introspection`. A way to circumvent this would be to alter one of the title to generate a different html anchor reference.

**Comment by rswgnu** [comment]
Yes, the anchors are fine for written communication.  I was thinking about verbal references where people were looking at the same version of the list, knowing that across time, the section numbers could change.

-- Bob
